webserver.py

```diff
@@ -46,6 +46,10 @@ class GameState(object):
     def __init__(self, json_data):
         self.data = json_data
 
+    def get_client_data(self):
+        del self.data['password']
```

Comment:
careful here.  You're actually removing the 'password' key from the data.  If you were to later save this game_state, you'd be storing a game without a password, so the user could never log back in.

Another problem is that you're exposing the underlying data itself.  It's not hard to imaging down the road having code like this:

```
client_data = game_state.get_client_data()
clients_seeds = client_data['seeds']
msg = ""
while client_seeds:
    msg += "You have %s %s seeds.\n" % client_seeds.pop_item()
print msg
...
client_data.save()
```

The above code would, as a side effect, empty the player's inventory.  The person writing the code probably wouldn't expect that to happen.


A strategy to deal with this is called a "defensive copy":  http://www.javapractices.com/topic/TopicAction.do?Id=15
Basically you make a copy of the data you care about and hand that out.  The caller can perform whatever changes they want on that copy without affecting your original.

python as a module `copy` with a method `deepcopy` that might come in handy here